### PR TITLE
Add documentation on how to use the API

### PR DESCRIPTION
**Pull request by ebaron**
This PR adds a new documentation file with a brief introduction to the FlightRecorder API provided by our operator, along with instructions on how to use it to list, create and download recordings.

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        "recordingRequests": [
+                {
+                        "name": "my-recording",
+                        "eventOptions": [ "jdk.SocketRead:enabled=true", "jdk.SocketWrite:enabled=true" ],
+                        "duration": "30s"
+                }
+        ]
+    },
+    "status": {
+        "recordings": [],
```
Comment (andrewazores) [inline API.md]: Looks like the JSON lost some braces and such here

File: API.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    },
+    "status": {
+        "recordings": [
+            {
+                "active": true,
+                "duration": "30s",
+                "name": "my-recording",
+                "startTime": "2020-01-23T15:55:04Z"
+            }
+        ],
```
Comment (andrewazores) [inline API.md]: Same here, looks like some missing braces

File: API.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+```
+
+You'll need to pass your bearer token with the curl request. (You may also need -k if your test cluster uses a self-signed certificate)
+```shell
+$ curl -k -H "Authorization: Bearer $(oc whoami -t)" https://containerjfr-default.apps-crc.testing:443/recordings/172-30-157-32_my-recording_20200123T155535Z.jfr my-recording.jfr
```
Comment (andrewazores) [inline API.md]: `oc whoami -t | base64` I think

File: API.md (hunk 1 above)
Comment (ebaron) [reply API.md]: This was actually cropped to only show the relevant parts of the JSON. I'll add the missing closing braces at least.

File: API.md (hunk 3 above)
Comment (ebaron) [reply API.md]: Is Base64 encoding for the web server as well or just for commands over WebSockets? Using version 0.13.0, I only succeed without the encoding.

Comment (andrewazores) [reply API.md]: Ah, you're right - whatever charset is used by some auth system's tokens is probably a permissible subset of what's allowed in HTTP headers, so I didn't add base64 encode/decode there. But now I'm questioning if that was a strong enough assumption. Anyway, your example reflects the actual current state of things, so it's good.

File: API.md (hunk 1 above)
Comment (ebaron) [reply API.md]: This should be fixed now.

File: API.md (hunk 2 above)
Comment (ebaron) [reply API.md]: Should also be fixed.

**Pull request merged**